pkg/cstor/pool/v1alpha2/import.go

```diff
@@ -0,0 +1,117 @@
+package v1alpha2
+
+import (
+	"github.com/golang/glog"
+	"github.com/openebs/maya/cmd/cstor-pool-mgmt/controller/common"
+	"github.com/openebs/maya/cmd/cstor-pool-mgmt/volumereplica"
+	apis "github.com/openebs/maya/pkg/apis/openebs.io/v1alpha1"
+	api "github.com/openebs/maya/pkg/apis/openebs.io/v1alpha2"
+	zfs "github.com/openebs/maya/pkg/zfs/cmd/v1alpha1"
+)
+
+// Import will import pool for given CSP object.
+// It will also set `cachefile` property for that pool
+// if it is mentioned in object
+func Import(csp *api.CStorNPool) (string, bool, error) {
+	ret, err := zfs.NewPoolImport().
+		WithCachefile(csp.Spec.PoolConfig.CacheFile).
+		WithPool(PoolName(csp)).
+		Execute()
+	if err != nil {
+		glog.Errorf("Failed to import pool : %s : %s", ret, err.Error())
+		// We return error as nil because pool doesn't exist
+		return "", false, nil
+	}
+
+	// We imported pool successfully
+	// Let's set cachefile for this pool, if it is provided in csp object
+	if len(csp.Spec.PoolConfig.CacheFile) != 0 {
+		if _, err := zfs.NewPoolSProperty().
+			WithProperty("cachefile", csp.Spec.PoolConfig.CacheFile).
+			Execute(); err != nil {
+			//TODO, If cachefile set failed, do we need to return status as offline?
+			glog.Errorf("Failed to set cachefile for pool {%s} : %s", PoolName(csp), err.Error())
+			common.SyncResources.IsImported = false
+			return string(apis.CStorPoolStatusOffline), true, err
+		}
+		glog.Infof("Set cachefile successful for pool {%s}", PoolName(csp))
+	}
+
+	// TODO: audit required
+	// GetVolumes is called because, while importing a pool, volumes corresponding
+	// to the pool are also imported. This needs to be handled and made visible
+	// to cvr controller.
+	common.InitialImportedPoolVol, err = volumereplica.GetVolumes()
+	if err != nil {
+		common.SyncResources.IsImported = false
+		return string(apis.CStorPoolStatusOffline), true, err
+	}
+
+	glog.Infof("Import Pool with cachefile successful: %v", string(csp.GetUID()))
+	common.SyncResources.IsImported = true
```

Comment:
setting this flag here doesn't make sense.. 
this also need to be reset or set to false if pool import failed in above steps

Reply:
its better to cleanup the code depending on this variable